q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,332 @@
+        parsed_rules : dict
+            Parsed variable's rules.
+
+        """
+        with open(variable_rules_fp) as handle:
+            try:
+                parsed_rules = yaml.load(handle, Loader=yaml.FullLoader)
+                return parsed_rules
+            except yaml.YAMLError:
+                raise IOError("Something is wrong with the .yml rule file.")
+            except yaml.scanner.ScannerError:
+                raise IOError("Something is wrong with the .yml rule file.")
+            except UnicodeDecodeError:
+                raise IOError("Something is wrong with the .yml rule file.")
+
+    def normalize(self, variable: str, input_column: pd.Series) -> pd.Series:
+        """
+        Dispatch the metadata column to be curated to
+        the functions that will be applied to properly process
+        the current variable according to the current set of
+        rules for it.
+
+        Parameters
+        ----------
+        variable : str
+            Name of the variable which rules are parsed and
+            that is the yaml file name without the .yml extension.
+        input_column : pd.Series
+            Metadata column to be curated (that may already got
+            through this function).
+
+        Returns
+        -------
+        output_column : pd.Series
+            Curated (or being curated) metadata column.
+
+        """
+        pass
+
+
+class RulesCollection(object):
+    """Collect all the rules for all the variables.
+
+    Parameters
+    ----------
+    variables_rules_dir : str
+        Path to the folder containing the variables rules in yaml.
+
+    Attributes
+    ----------
+    variables_rules_files : list
+        Paths to the yaml rules files.
+    warnings : WarningsCollection instance
+        Collection of warnings encountered during either the parsing
+        of the variables rules, or during the application the rules.
+    errors : ErrorsCollection instance
+        Collection of errors encountered during either the parsing
+        of the variables rules, or during the application the rules.
+
+    Methods
+    -------
+    check_variables_rules
+        Fills `rules` attribute with correctly formatted rules
+        for the current variable, or raise user-defined exception.
+    _check_variables_rules_dir
+        Performs checks on rules paths input. Either raise Exceptions,
+        or return the list `variables_rules_files` containing the paths
+        to the existing yml rules files.
+    parse_variables_rules
+        Parse the yml rules files, one variable at a time and returns the
+        dict `variables_rules` that for each variable (keys) leads to an
+        instance of the `Rules()` class. This class performs rule-formatting
+        checks and fills a rules dict structure.
+
+    Raises
+    ------
+    IOError
+        If directory passed to --p-rules-dir does not exist.
+    FileNotFoundError
+        If directory passed to --p-rules-dir contains no .yml file.
```

Comment:
Make sure to remove --p-rules-dir.

Reply:
accounted for - will be part of upcoming commit